# Java IOException

> 原文:[https://www.educba.com/java-ioexception/](https://www.educba.com/java-ioexception/)

![Java IOException](../Images/6ccb5925cc913f37beda6eb768ecd622.png)

<noscript><img class="alignnone size-full wp-image-369406" src="../Images/6ccb5925cc913f37beda6eb768ecd622.png" alt="Java IOException" width="900" height="500" srcset="https://cdn.educba.com/academy/wp-content/uploads/2020/05/Java-IOException.jpg 900w, https://cdn.educba.com/academy/wp-content/uploads/2020/05/Java-IOException-300x167.jpg 300w, https://cdn.educba.com/academy/wp-content/uploads/2020/05/Java-IOException-768x427.jpg 768w" sizes="(max-width: 900px) 100vw, 900px" data-original-src="https://cdn.educba.com/academy/wp-content/uploads/2020/05/Java-IOException.jpg"/></noscript>

## Java IOException 简介

Java IOException 或 IOException 通常通过数据流、文件系统和序列化等为一些系统输入和输出提供帮助。这是 java.util.scanner java 类的方法，它实际上返回 IOException，并且是某个扫描器的底层可读。仅当实际上不存在这样的异常时，此 IOException 方法才会返回 NULL 值。

**语法:**

<small>网页开发、编程语言、软件测试&其他</small>

```
Public IOException ioException()
```

**io exception 的返回值:**

IOException 方法/函数将返回最后一个异常，它实际上是由扫描器的 readable/readables 抛出的。

### IOException 如何在 Java 中工作？

Java 语言的 IOException 通过为数据流、序列化和文件系统中的一些输入和输出提供一些帮助来工作。IoException()方法/函数通过返回最后抛出的扫描程序的 IOException 的底层可读概念来工作。仅当根本没有异常可用时，IOException()方法/函数才会返回 NULL 值。

**如何避免 IOException？**

使用 Java 编程语言的 Try/Catch 概念，我们可以处理 IOException。这是一个避免 IOException 的概念。

### Java 中 IOException 的构造函数

通常，构造函数在对象创建后帮助初始化对象。它在语法上与特定的方法/函数相似/相同，但是构造函数有一些不同。它与没有返回类型的类具有相同的。实际上，不需要任何显式调用的构造函数，这些将在实例化时自动调用。这些构造函数抛出一些异常。

**1。IOException():** 这是一个普通的构造函数，它构造了一个新的 IOException 及其堆栈跟踪，它将被填充。

**2。io exception(Throwable):**Throwable 构造函数帮助构造一个新的类实例及其详细的原因，这将被填充。这里“Throwable”参数是原因。

**3。IOException(String):**io exception()的字符串构造函数帮助构造一个新的 io exception 及其堆栈跟踪和详细消息:filled。

**4。IOException(IntPtr，JniHandleOwnership):** 这个构造函数帮助创建一些 JNI 对象的托管表示，运行时调用它们。IntPtr 将包含一些用于对象引用目的的 Java 本地接口(JNI)。JniHandleOwnership 参数指示处理 javaReference。

**5。IOException(String，Throwable):** 构造函数帮助构造一个新的类实例，并附带一些详细的消息和原因填充。这里，String 参数是消息，Throwable 参数是原因。

### Java IOException 的实现示例

以下是 Java IOException 的示例:

#### 示例#1

这是在不使用任何参数的情况下，在 java 编程语言中实现某些扫描器类的 ioException()方法/函数的说明的 Java 程序示例。这里首先导入 java util 来使用 java 编程语言的所有库。然后使用异常抛出概念创建 pavankumarsake1 类。然后用字符串值创建 s1 字符串变量。然后在 s1 字符串变量的帮助下创建 scanner1 变量。然后创建 scanner.nextLine()打印新行，然后使用 scanner.ioException()检查 IO 异常是否存在。然后将使用 scanner.close()函数关闭 scanner。

**代码:**

```
import java.util.*;
public class pavansake1 {
public static void main(String[] args)
throws Exception
{
System.out.println(" \n ");
String s1 = "Hey! I'am from EDUCBA";
Scanner scanner1 = new Scanner(s1);
System.out.println("" + scanner1.nextLine());
System.out.println("" + scanner1.ioException());
scanner1.close();
}
}
```

**输出:**

![Java IO Exception Example 1](../Images/dcc6177c991326e84550d8668762b728.png)

<noscript><img class="alignnone size-full wp-image-369020" src="../Images/dcc6177c991326e84550d8668762b728.png" alt="Java IO Exception Example 1" width="166" height="70" data-original-src="https://cdn.educba.com/academy/wp-content/uploads/2020/05/Java-IO-Exception-Example-1.png"/></noscript>

#### 实施例 2

这是一个 Java 示例，说明了 ioException()方法及其在 Java 编程语言中 Scanner 类的一些构造函数，但没有提到参数。这里首先导入 java.util.*来导入库的所有函数。然后通过抛出异常概念创建公共类“pavansake1”。在抛出异常时，用某个字符串值创建一个字符串 s11。然后，在 s11 变量的帮助下创建扫描仪变量，以将其用作替代的类似变量。然后 ioException()变量用于显示是否有 IO 异常。

**代码:**

```
import java.util.*;
public class pavansake1 {
public static void main(String[] argv)
throws Exception
{
System.out.println("\n");
String s11 = "EDUCBA EDUCBA!!!";
Scanner scanner11 = new Scanner(s11);
System.out.println("" + scanner11.nextLine());
System.out.println("" + scanner11.ioException());
scanner11.close();
}
}
```

**输出:**

![Java IO Exception Example 2](../Images/d51e8bd01402c891d99f5d66076c2a17.png)

<noscript><img class="alignnone size-full wp-image-369022" src="../Images/d51e8bd01402c891d99f5d66076c2a17.png" alt="Java IO Exception Example 2" width="143" height="76" data-original-src="https://cdn.educba.com/academy/wp-content/uploads/2020/05/Java-IO-Exception-Example-2.png"/></noscript>

#### 实施例 3

这是实现 IOException 以及其中的构造函数的示例，但是由于一个问题，该示例将提供一个编译错误。这里创建了一个类“Employee1 ”,它的构造函数将抛出一个 IOException，它正在实例化不处理我们的异常的类。所以编译会导致编译运行时错误。有些 Java 库是先导入，然后创建私有类和公有类来处理 IOException，但是这里没有处理 Exception，所以出现了错误。

**代码:**

```
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
class Employee1{
private String name1;
private int age1;
File empFile1;
Employee1(String name1, int age1, String empFile1) throws IOException{
this.name1 = name1;
this.age1 = age1;
this.empFile1 = new File(empFile1);
new FileWriter(empFile1).write("Employee name is "+name1+"and age is "+age1);
}
public void display(){
System.out.println("Name: "+name1);
System.out.println("Age: "+age1);
}
}
public class ConstructorExample1 {
public static void main(String args[]) {
String filePath1 = "samplefile.txt";
Employee emp1 = new Employee("pksake", 25, filePath);
}
}
```

**输出:**

![Constructor Example 3](../Images/166b080a0dba9b58560a54c7a3cd12c2.png)

<noscript><img class="alignnone wp-image-369025 size-full" src="../Images/166b080a0dba9b58560a54c7a3cd12c2.png" alt="Constructor Example 3" width="594" height="69" srcset="https://cdn.educba.com/academy/wp-content/uploads/2020/05/Java-IO-Exception-Example-3.png 594w, https://cdn.educba.com/academy/wp-content/uploads/2020/05/Java-IO-Exception-Example-3-300x35.png 300w" sizes="(max-width: 594px) 100vw, 594px" data-original-src="https://cdn.educba.com/academy/wp-content/uploads/2020/05/Java-IO-Exception-Example-3.png"/></noscript>

#### 实施例 4

这个例子与例 3 非常相似。为了让示例 3 起作用，我们将该行的实例化包装在 try-catch 或 throw 异常中。

**代码:**

```
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
class Employee1{
private String name1;
private int age1;
File empFile1;
Employee1(String name1, int age1, String empFile1) throws IOException{
this.name1 = name1;
this.age1 = age1;
this.empFile1 = new File(empFile1);
new FileWriter(empFile1).write("Employee name is "+name1+"and age is "+age1);
}
public void display(){
System.out.println("Name: "+name1);
System.out.println("Age: "+age1);
}
}
public class Employee11 {
public static void main(String args[]) {
String filePath1 = "file1.txt";
Employee1 emp1 = null;
try{
emp1 = new Employee1("pksake", 26, filePath1);
}catch(IOException ex1){
System.out.println("The specific file will not be found");
}
emp1.display();
}
}
```

**输出:**

![try-catch Example 4](../Images/316b0122f8d46807bcf9c09633648dc3.png)

<noscript><img class="alignnone wp-image-369026 size-full" src="../Images/316b0122f8d46807bcf9c09633648dc3.png" alt="try-catch Example 4" width="122" height="64" data-original-src="https://cdn.educba.com/academy/wp-content/uploads/2020/05/Java-IO-Exception-Example-4.png"/></noscript>

### 结论

我希望你已经了解了 Java IOException 的定义及其语法和解释，IOException 在 Java 编程语言及其构造函数中是如何工作的，Java IOException 的例子，以及如何避免 IOException 等。

### 推荐文章

这是 Java IOException 的指南。这里我们讨论 Java IOException 的介绍以及它如何与不同的构造函数一起工作。您也可以浏览我们推荐的其他文章，了解更多信息——

1.  [Java varargs](https://www.educba.com/java-varargs/)
2.  [Java max()](https://www.educba.com/java-max/)
3.  [Java 中的 copy()](https://www.educba.com/copy-in-java/)
4.  [Java 8 中的可选类](https://www.educba.com/optional-class-in-java-8/)

<footer class="entry-footer">

<aside class="sidebar sidebar-primary widget-area" role="complementary" aria-label="Primary Sidebar">Primary Sidebar</aside>

</footer>